commit 8ea13011bbb0fcf232c1cdd5f7763ab2ac1e7455
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Tue Feb 7 01:13:39 2017 +0100

    Consolidate {KF5,QT5}_MIN_VERSION, mark KF5Plasma as RUNTIME dep
    
    In addition, sort and cleanup deps
    
    REVIEW: 129930

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b053562..39c45e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,36 +19,40 @@ include(KDECMakeSettings)
 include(FeatureSummary)
 include(ECMInstallIcons)
 
+set(QT_MIN_VERSION "5.4.0")
+set(KF5_MIN_VERSION "5.9.0")
+
 # Qt5 modules
-find_package(Qt5Core 5.4.0 REQUIRED)
-find_package(Qt5Gui 5.4.0 REQUIRED)
-find_package(Qt5Widgets 5.4.0 REQUIRED)
-find_package(Qt5Qml 5.4.0 REQUIRED)
-find_package(Qt5Test 5.4.0 REQUIRED)
-find_package(Qt5Network 5.4.0 REQUIRED)
-find_package(Qt5PrintSupport 5.4.0 REQUIRED)
+find_package(Qt5Core ${QT_MIN_VERSION} REQUIRED)
+find_package(Qt5Gui ${QT_MIN_VERSION} REQUIRED)
+find_package(Qt5Network ${QT_MIN_VERSION} REQUIRED)
+find_package(Qt5PrintSupport ${QT_MIN_VERSION} REQUIRED)
+find_package(Qt5Qml ${QT_MIN_VERSION} REQUIRED)
+find_package(Qt5Test ${QT_MIN_VERSION} REQUIRED)
+find_package(Qt5Widgets ${QT_MIN_VERSION} REQUIRED)
 
 # KF5 modules
-find_package(KF5Config 5.9.0 REQUIRED)
-find_package(KF5Auth 5.9.0 REQUIRED)
-find_package(KF5DocTools 5.9.0 REQUIRED)
-find_package(KF5IconThemes 5.9.0 REQUIRED)
-find_package(KF5WidgetsAddons 5.9.0 REQUIRED)
-find_package(KF5I18n 5.9.0 REQUIRED)
-find_package(KF5Completion 5.9.0 REQUIRED)
-find_package(KF5CoreAddons 5.9.0 REQUIRED)
-find_package(KF5Solid 5.9.0 REQUIRED)
-find_package(KF5KIO 5.9.0 REQUIRED)
-find_package(KF5Notifications 5.9.0 REQUIRED)
-find_package(KF5XmlGui 5.9.0 REQUIRED)
-find_package(KF5JobWidgets 5.9.0 REQUIRED)
-find_package(KF5Wallet 5.9.0 REQUIRED)
-find_package(KF5DBusAddons 5.9.0 REQUIRED)
-find_package(KF5Parts 5.9.0 REQUIRED)
-find_package(KF5ConfigWidgets 5.9.0 REQUIRED)
-find_package(KF5Notifications 5.9.0 REQUIRED)
-find_package(KF5WindowSystem 5.9.0 REQUIRED)
-find_package(KF5Plasma 5.9.0 REQUIRED)
+find_package(KF5Auth ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5Completion ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5Config ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5ConfigWidgets ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5CoreAddons ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5DBusAddons ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5DocTools ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5I18n ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5IconThemes ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5JobWidgets ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5KIO ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5Notifications ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5Parts ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5Solid ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5Wallet ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5WidgetsAddons ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5WindowSystem ${KF5_MIN_VERSION} REQUIRED)
+find_package(KF5XmlGui ${KF5_MIN_VERSION} REQUIRED)
+
+find_package(KF5Plasma ${KF5_MIN_VERSION})
+set_package_properties(KF5Plasma PROPERTIES TYPE RUNTIME)
 
 # Make sure that Smb4K builds when several custom targets
 # with the same name exist (happens in the po directory).
